library/core/src/main/java/com/google/android/exoplayer2/upstream/cache/SimpleCache.java

```diff
@@ -837,7 +837,7 @@ private static long parseUid(String fileName) {
   }
 
   private static void createCacheDirectories(File cacheDir) throws CacheException {
-    if (!cacheDir.mkdirs()) {
+    if (!cacheDir.mkdirs() && !cacheDir.exists()) {
```

Comment:
Would it be a bit better to use:

`if (!cacheDir.mkdirs() && (!cacheDir.exists() || !cacheDir.isDirectory())) {`